persistence/proto/src/main/scala/net/liftweb/proto/ProtoUser.scala

```diff
@@ -864,8 +864,10 @@ trait ProtoUser {
       }
     }
 
+    val emailElemId = nextFuncName
+    S.appendJs(Focus(emailElemId))
     val bind =
-      ".email" #> FocusOnLoad(<input type="text" name="username"/>) &
+      ".email" #> <input type="text" name="username" id={emailElemId}/> &
```

Comment:
yea good points, as I am poking around in this code I will take a look at minimizing the node replacements.
 
Lets take a look at what I am (we are) trying to accomplish in this PR
**Goal:**
  1.  Avoid Inline JS, hens replace or modify the FocusOnLoad call used in ProtoUser.
  2.  Somehow preserve the original behaviour although we modify or remove the use of FocusOnLoad.

**The original behaviour** that we try to preserve is in short  
User clicks the "Login" menu and the page loads and sets focus on the email input element so that **without any future user interaction** (click or key strokes) the user can start typing his/here email address in the email input field. 

**The road so far:**
I decided that my original approacher to modify the FocusOnLoad object was a bad idea case it introduced a unnecessary side effect and Tim showed a solution that I agreed was a less intrusive solution than the one I had come up with. 
Using the solution Tim suggested dose satisfied the goals (1,2) stated above but the suggestions you made to "let the attribute extraction stuff take care of it" (lets call this goal 3) is appealing so while we are at it we should take a stab on it. 
The solutions I have been trying has been unsuccessful and unfortunately the suggestion to use onclick "this.focus()" dose not satisfy this goal (3) either.  

@Shadowfiend  (or anyone else) any ideas ?